pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/html.py

```diff
@@ -214,11 +214,15 @@ def ordered_list_close(self, token: Token, tokens: Sequence[Token], i: int) -> s
         self._ordered_list_nesting -= 1;
         return "</ol></div>"
     def example_open(self, token: Token, tokens: Sequence[Token], i: int) -> str:
-        if id := token.attrs.get('id'):
-            return f'<a id="{escape(cast(str, id), True)}" />'
-        return ""
+        if id := cast(str, token.attrs.get('id', '')):
+            id = f'id="{escape(id, True)}"' if id else ''
+        return f'<div class="example"><a {id} />'
```

Comment:
Why not put the id attribute to the `<div>`? The anchor mode was previously done with `name` attribute, which is [apparently deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#name) and the HTML spec considers `<a>` without `href` a [different use case](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element:the-a-element-3) too.